File: test-summary-lib.js
/**
 * Library for testing type summary output.
 */

function greet(name, greeting) {
  return { message: `${greeting}, ${name}!`, name, greeting };
}

function add(a, b) {
  return { result: a + b, a, b };
}

function toUpper(text) {
  return { original: text, upper: text.toUpperCase() };
}

module.exports = { greet, add, toUpper };

File: test_summary_lib.py
"""Library for testing type summary output."""


def greet(name, greeting):
    return {"message": f"{greeting}, {name}!", "name": name, "greeting": greeting}


def add(a, b):
    return {"result": a + b, "a": a, "b": b}


def to_upper(text):
    return {"original": text, "upper": text.upper()}
